source/client/client_worker_impl.cc

```diff
@@ -17,18 +22,21 @@ ClientWorkerImpl::ClientWorkerImpl(Envoy::Api::Api& api, Envoy::ThreadLocal::Ins
                                    Envoy::Stats::Store& store, const int worker_number,
                                    const Envoy::MonotonicTime starting_time,
                                    Envoy::Tracing::HttpTracerPtr& http_tracer)
-    : WorkerImpl(api, tls, store), worker_scope_(store_.createScope("cluster.")),
+    : WorkerImpl(api, tls, store), termination_predicate_factory_(termination_predicate_factory),
+      sequencer_factory_(sequencer_factory), worker_scope_(store_.createScope("cluster.")),
       worker_number_scope_(worker_scope_->createScope(fmt::format("{}.", worker_number))),
       worker_number_(worker_number), starting_time_(starting_time), http_tracer_(http_tracer),
       request_generator_(request_generator_factory.create()),
       benchmark_client_(benchmark_client_factory.create(
           api, *dispatcher_, *worker_number_scope_, cluster_manager, http_tracer_,
           fmt::format("{}", worker_number), *request_generator_)),
-      termination_predicate_(
-          termination_predicate_factory.create(time_source_, *worker_number_scope_, starting_time)),
-      sequencer_(sequencer_factory.create(time_source_, *dispatcher_, starting_time,
-                                          *benchmark_client_, *termination_predicate_,
-                                          *worker_number_scope_)) {}
+      phase_(std::make_unique<PhaseImpl>(
+          "main",
```

Comment:
No this shouldn’t matter for now, it will only show up in logs but otherwise is not visible externally.

Reply:
Just to clarify, does that mean this is temporary code that will be replaced later?  Or that the use of main here is a secret phase that outside users won't need to know about?  If it's the latter, I'm concerned about name collision, but if it's the first, then agree there's no concern here.

Reply:
It's temporary code. This will be replaced with a generic mechanism that translates phases into configuration. So the id chosen here can be an arbitrary value, perhaps I should have better called it `foo` or some such to avoid confusion.